From 08d9c0d6656820250b0844638c5471f3335914f2 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 24 Jan 2014 09:21:30 +0000
Subject: [PATCH 30/38] Add glibc 2.15 patch from OpenWrt for cross compiling

Url
https://dev.openwrt.org/browser/trunk/toolchain/glibc/patches/2.14/100-fix_cpuid_h_check.patch?rev=30478

It's an autoconf issue with an in-complete sysroot.
---
 .../170-100-fix_cpuid_h_check-from-OpenWrt.patch   | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)
 create mode 100644 patches/glibc/2.15/170-100-fix_cpuid_h_check-from-OpenWrt.patch

diff --git a/patches/glibc/2.15/170-100-fix_cpuid_h_check-from-OpenWrt.patch b/patches/glibc/2.15/170-100-fix_cpuid_h_check-from-OpenWrt.patch
new file mode 100644
index 0000000..255c87f
--- /dev/null
+++ b/patches/glibc/2.15/170-100-fix_cpuid_h_check-from-OpenWrt.patch
@@ -0,0 +1,22 @@
+--- a/sysdeps/i386/configure.in
++++ b/sysdeps/i386/configure.in
+@@ -2,7 +2,7 @@
+ # Local configure fragment for sysdeps/i386.
+ 
+ AC_HEADER_CHECK([cpuid.h], ,
+-  [AC_MSG_ERROR([gcc must provide the <cpuid.h> header])])
++  [AC_MSG_ERROR([gcc must provide the <cpuid.h> header])], [])
+ 
+ AC_CACHE_CHECK(if -g produces usable source locations for assembler-with-cpp,
+ 	       libc_cv_cpp_asm_debuginfo, [dnl
+--- a/sysdeps/i386/configure.orig	2014-01-24 09:13:37.232015700 +0000
++++ b/sysdeps/i386/configure	2014-01-24 09:14:32.788193300 +0000
+@@ -632,7 +632,7 @@
+ done
+ 
+ 
+-ac_fn_c_check_header_mongrel "$LINENO" "cpuid.h" "ac_cv_header_cpuid_h" "$ac_includes_default"
++ac_fn_c_check_header_mongrel "$LINENO" "cpuid.h" "ac_cv_header_cpuid_h" ""
+ if test "x$ac_cv_header_cpuid_h" = xyes; then :
+ 
+ else
-- 
2.0.1

